src/test/java/org/beanpod/switchboard/controller/DecoderControllerTest.java

```diff
@@ -130,8 +142,34 @@ final void testDeleteDecoder(){
     //When a decoder is unavailable in the DB
     @Test
     final void testDeleteDecoderNotExisting(){
-        assertThrows(DeviceNotFoundException.class, () -> {
+        assertThrows(ExceptionType.DeviceNotFoundException.class, () -> {
             decoderController.deleteDecoder("Not Available decoder");
         }, "DeviceNotFoundException should have been thrown.");
     }
+
+
+    //When a encoder is available in the DB
+    @Test
+    final void testUpdateDecoder(){
+
+        when(decoderService.findDecoder("1")).thenReturn(java.util.Optional.of(decoder1));
```

Comment:
can't we just import optional & get rid of the `java.util.`?

Reply:
yes, fixed